### PR TITLE
[WIP] Add Skew function to GeoM

### PR DESCRIPTION
**Pull request by neganp**
This is my initial attempt at adding a Skew function to `GeoM`. If you have ideas or preferences about how I should test the new method, please let me know and I'll do my best to implement them. Otherwise I'll do my best to figure it out on my own.

### CODE REVIEW

**Comment by neganp** [comment]
This would close #627

File: geom.go [hunk 1]
```diff
@@ -162,6 +162,25 @@ func (g *GeoM) Rotate(theta float64) {
 	g.ty = ty
 }
 
+// Skew the image.
```
Comment (hajimehoshi) [inline geom.go]: `Skew skews the matrix by (skewX, skewY). The unit is radian.

File: geom.go [hunk 2]
```diff
@@ -162,6 +162,25 @@ func (g *GeoM) Rotate(theta float64) {
...
+func (g *GeoM) Skew(skewX, skewY float64) {
+
```
Comment (hajimehoshi) [inline geom.go]: Remove this empty line.

File: geom_test.go [hunk 3]
```diff
@@ -286,5 +286,6 @@ func BenchmarkGeoM(b *testing.B) {
 		m.Translate(10, 20)
 		m.Scale(2, 3)
 		m.Rotate(math.Pi / 2)
+		m.Skew(2, 3)
```
Comment (hajimehoshi) [inline geom_test.go]: Instead of adding calling at a benchmark, could you add a unit test to check the result is correct?

File: geom.go [hunk 4]
```diff
@@ -162,6 +162,25 @@ func (g *GeoM) Rotate(theta float64) {
...
+	sx64 := math.Tan(skewX)
+	sy64 := math.Tan(skewY)
+	sx, sy := float32(sx64), float32(sy64)
+
+	a := g.c * sy
+	b := (g.d_1 + 1) * sy
+	c := (g.a_1 + 1) * sx
+	d := g.b * sx
```
Comment (hajimehoshi) [inline geom.go]: Is this calculation really correct? I expect this multiplying:

```
| 0          tan(skewX) 0 || a b tx |
| tan(skewY) 0          0 || c d ty |
| 0          0          1 || 0 0 1  |
```

Then tx and ty are also affected?

Comment (hajimehoshi) [reply geom.go]: Sorry but

```
| 1          tan(skewX) 0 || a b tx |
| tan(skewY) 1          0 || c d ty |
| 0          0          1 || 0 0 1  |
```

is the correct calculation I think. (https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/geom/Matrix.html includes a wrong skew matrix?)

Comment (neganp) [reply geom.go]: You're absolutely right; I thought that result looked fishy! Thanks for the catch, and sorry for my confusion. I'll fix this today.

**Comment by neganp** [comment]
I think that should be everything, and it should rebase cleanly onto master. If you'd prefer the PR to have just the commits for the `Skew` method, I can pull those out and re-submit (not sure if you prefer merge commits or rebase).

**Comment by hajimehoshi** [comment]
Now this PR includes changes that are not related to skewing. Could you merge or rebase this? You don't have to squash them (I'll do). Thanks!

**Comment by neganp** [comment]
I've taken out the extra commits and am working on the other changes you suggested.

File: geom_test.go [hunk 5]
```diff
@@ -279,6 +287,48 @@ func TestGeoMIsInvert(t *testing.T) {
...
+	outp.SetElement(0, 1, b)
+	outp.SetElement(0, 2, tx)
+	outp.SetElement(1, 0, c)
+	outp.SetElement(1, 1, d)
+	outp.SetElement(1, 2, ty)
+	return outp
+}
+
+func TestGeomSkew(t *testing.T) {
+
```
[the remark this answers is not in the document]
Comment (hajimehoshi) [reply geom_test.go]: ping

File: geom_test.go [hunk 6]
```diff
@@ -279,6 +286,55 @@ func TestGeoMIsInvert(t *testing.T) {
...
+	testSkew := func(skewX, skewY float64, input, expected GeoM) {
+		input.Skew(skewX, skewY)
+		for i := 0; i < 3; i++ {
+			for j := 0; j < 3; j++ {
+				got := input.Element(i, j)
+				want := expected.Element(i, j)
+				if want != got {
+					t.Errorf("Geom{}.Skew(1, 0): got %s, want: %s", input.String(), expected.String())
```
Comment (hajimehoshi) [inline geom_test.go]: What is `(1, 0)`?

**Comment by neganp** [comment]
Great! Thanks so much for all your patience and help!

**Pull request merged**